app/models/partition/script_runner.rb

```diff
@@ -54,45 +57,25 @@ def status
 
     def debounce_runner(path, type:)
       @mutex.synchronize do
-        # Set the next time the debouncer can run according to the minimum period
-        @debouncing[:after][type] = Process.clock_gettime(Process::CLOCK_MONOTONIC) + FlightScheduler.app.config.min_debouncing
-        task = @debouncing[:tasks][type]
-        if task && !task&.finished?
-          Async.logger.debug "Skipping partition '#{partition.name}' #{type} script as it is scheduled to run"
-          return
-        end
+        key = SHARED_DEBOUNCING[type]
+        now = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+        interval = @cooldown[key] - now
 
-        # Start the main script handler
-        t = Async do |task|
-          Async.logger.info "Scheduling partition '#{partition.name}' #{type} script to be ran"
-          start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
-          finish_time = start_time + FlightScheduler.app.config.max_debouncing
-
-          # Loop until the debouncing condition is met
-          loop do
-            current_time = Process.clock_gettime(Process::CLOCK_MONOTONIC)
-
-            # Exit if the minimum time has elapsed without an additional request
-            if current_time > @debouncing[:after][type]
-              break
-            # Wait the minimum time if it would not exceed the maximum
-            elsif finish_time > current_time + FlightScheduler.app.config.min_debouncing
-              task.sleep FlightScheduler.app.config.min_debouncing
-            # Wait out the minimum time period
-            elsif (diff = (finish_time - current_time).to_i) > 0
-              task.sleep diff
-            # Exit because the maximum time has been exceeded +/- one second
-            else
-              break
-            end
-          end
+        if interval < 0
+          # TODO: Change the configuration key
```

Comment:
Is there a reason this can't be done now?  Also what needs changing here?

Reply:
https://github.com/openflighthpc/flight-scheduler-controller/pull/52/commits/915ec90a091970ad3f06db625321f57f65af92af